src/Compilers/Core/MSBuildTaskTests/GenerateMSBuildAnalyzerConfigTests.cs

```diff
@@ -0,0 +1,257 @@
+            Assert.Equal(@"is_global = true
+", result);
+        }
+
+        [Fact]
+        public void PropertiesAreGeneratedInGlobalSection()
+        {
+            TaskItem property1 = new TaskItem("Property1", new Dictionary<string, string> { { "Value", "abc123" } });
+            TaskItem property2 = new TaskItem("Property2", new Dictionary<string, string> { { "Value", "def456" } });
+
+            GenerateMSBuildAnalyzerConfig configTask = new GenerateMSBuildAnalyzerConfig()
+            {
+                PropertyItems = new[] { property1, property2 }
+            };
+            configTask.Execute();
+
+            var result = configTask.ConfigFileContents;
+
+            Assert.Equal(@"is_global = true
+msbuild_property.Property1 = abc123
+msbuild_property.Property2 = def456
+", result);
+        }
+
+        [Fact]
+        public void ItemMetaDataCreatesSection()
+        {
+            TaskItem item1 = new TaskItem("c:\\file1.cs", new Dictionary<string, string> { { "ItemType", "Compile" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "abc123" } });
+
+            GenerateMSBuildAnalyzerConfig configTask = new GenerateMSBuildAnalyzerConfig()
+            {
+                MetadataItems = new[] { item1 }
+            };
+            configTask.Execute();
+
+            var result = configTask.ConfigFileContents;
+
+            Assert.Equal(@"is_global = true
+
+[c:\file1.cs]
+msbuild_item.Compile.ToRetrieve = abc123
+", result);
+        }
+
+        [Fact]
+        public void MutlipleItemMetaDataCreatesSections()
+        {
+            TaskItem item1 = new TaskItem("c:\\file1.cs", new Dictionary<string, string> { { "ItemType", "Compile" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "abc123" } });
+            TaskItem item2 = new TaskItem("c:\\file2.cs", new Dictionary<string, string> { { "ItemType", "Compile" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "def456" } });
+            TaskItem item3 = new TaskItem("c:\\file3.cs", new Dictionary<string, string> { { "ItemType", "AdditionalFiles" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "ghi789" } });
+
+            GenerateMSBuildAnalyzerConfig configTask = new GenerateMSBuildAnalyzerConfig()
+            {
+                MetadataItems = new[] { item1, item2, item3 }
+            };
+            configTask.Execute();
+
+            var result = configTask.ConfigFileContents;
+
+            Assert.Equal(@"is_global = true
+
+[c:\file1.cs]
+msbuild_item.Compile.ToRetrieve = abc123
+
+[c:\file2.cs]
+msbuild_item.Compile.ToRetrieve = def456
+
+[c:\file3.cs]
+msbuild_item.AdditionalFiles.ToRetrieve = ghi789
+", result);
+        }
+
+        [Fact]
+        public void DuplicateItemSpecsAreCombinedInSections()
+        {
+            TaskItem item1 = new TaskItem("c:\\file1.cs", new Dictionary<string, string> { { "ItemType", "Compile" }, { "MetadataName", "ToRetrieve" }, { "ToRetrieve", "abc123" } });
```

Comment:
>MetadataName [](start = 121, length = 12)

Just to confirm, "ItemType" and "MetadataName" are handled/provided by MSBuild infrastructure, so we don't have to worry about casing variations, correct?

Reply:
Yes.

---
In reply to: [419696372](https://github.com/dotnet/roslyn/pull/43617#discussion_r419696372) [](ancestors = 419696372)